src/Components/EnvironmentPicker/EnvironmentPickerManager.test.ts

```diff
@@ -0,0 +1,153 @@
+import { cleanup } from '@testing-library/react-hooks';
+import { AzureResourceTypes } from '../../Models/Constants/Enums';
+import {
+    MOCK_ENVIRONMENT,
+    MOCK_ENVIRONMENT_URL,
+    MOCK_STORAGE_ACCOUNT,
+    MOCK_STORAGE_ACCOUNT_URL,
+    MOCK_STORAGE_CONTAINER,
+    MOCK_STORAGE_CONTAINER_NAME,
+    MOCK_STORAGE_CONTAINER_URL
+} from './EnvironmentPicker.mock';
+import {
+    areResourceUrlsEqual,
+    getContainerDisplayText,
+    getContainerName,
+    getContainerNameFromUrl,
+    getEnvironmentDisplayText,
+    getResourceUrl,
+    getStorageAccountUrlFromContainerUrl,
+    getStorageAndContainerFromContainerUrl
+} from './EnvironmentPickerManager';
+
+afterEach(cleanup);
+
+describe('EnvironmentPickerManager', () => {
+    test('ADT instance/environment name is displayed properly', () => {
+        const displayText1 = getEnvironmentDisplayText(MOCK_ENVIRONMENT);
+        const displayText2 = getEnvironmentDisplayText(MOCK_ENVIRONMENT_URL);
+
+        expect(displayText1).toBe('testADTInstance');
+        expect(displayText2).toBe('testADTInstance');
+    });
+
+    test('Storage container name is displayed properly', () => {
+        const displayText1 = getContainerDisplayText(
+            MOCK_STORAGE_CONTAINER,
+            MOCK_STORAGE_ACCOUNT
+        );
+        const displayText2 = getContainerDisplayText(
+            MOCK_STORAGE_CONTAINER_NAME,
+            MOCK_STORAGE_ACCOUNT_URL
+        );
+
+        expect(displayText1).toBe('teststorageaccount/teststoragecontainer');
+        expect(displayText2).toBe('teststorageaccount/teststoragecontainer');
+    });
+
+    test('Getting resource url properly', () => {
```

Comment:
in my ideal world, each test validates a single operation/functionality rather than multiple variations. I find that it helps to self document the scenario since each gets it's own test name and if one condition starts failing you know exactly what was being validated rather than that _something_ about this function has changed. 
What are your thoughts?

Reply:
I am not a testing expert but to my mind this test method is testing the `getResourceUrl` method. I just tried to exemplify different inputs for this method to use its parameters as different as possible (e.g. for different resource types like resource url of adt instace/storage account/container). So it is testing a single functionality with different inputs.